python/pyarrow/array.pxi

```diff
@@ -1253,6 +1253,16 @@ cdef class Array(_PandasConvertible):
         options = _pc().NullOptions(nan_is_null=nan_is_null)
         return _pc().call_function('is_null', [self], options)
 
+    def is_nan(self):
+        """
+        Return BooleanArray indicating the nan values.
+
+        Returns
+        -------
+        array : boolean Array, where null values are None
```

Comment:
What does this "where null values are None" mean exactly?

Reply:
That a null value, will result in a Python None:
https://github.com/apache/arrow/blob/df625f2abf7f460b9935503998c075ad9f3c5e62/python/pyarrow/tests/test_compute.py#L1639-L1643

Reply:
Ah, OK, I didn't get that at first. Now, I don't think that is something we need to mention here in this docstring (that's true for _all_ functions that return arrow data). And it's also only when converting to native Python objects that this happens, the actual result here is a pyarrow.array, and that doesn't use None (so the explanation would be relevant for a method like `Array.to_pylist()`.

The reason you see it in the tests, is because we allow you to use None to specify a null value when constructing an array (but the None gets converted to a proper "null").

Reply:
@jorisvandenbossche Okay, that's fine with me as well. It was not directly obvious for me, but it makes sense now I'm aware of it.